web/src/test/java/org/cbioportal/web/ReferenceGenomeGeneControllerTest.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration
+@ContextConfiguration("/applicationContext-web-test.xml")
+@Configuration
+public class ReferenceGenomeGeneControllerTest {
+    
+    public static final String CYTOBAND_1 = "cytoband_1";
+    public static final int LENGTH_1 = 100;
+    public static final String CHROMOSOME_1 = "chromosome_1";
+    public static final String CYTOBAND_2 = "cytoband_2";
+    public static final int LENGTH_2 = 200;
+    public static final String CHROMOSOME_2 = "chromosome_2";
+    public static final int REFERENCE_GENOME_ID = 1;
+    public static final int ENTREZ_GENE_ID_1 = 1;
+    public static final int ENTREZ_GENE_ID_2 = 2;
+    
+    @Autowired
+    private WebApplicationContext wac;
+
+    @Autowired
+    private ReferenceGenomeGeneService referenceGenomeGeneService;
+    
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private MockMvc mockMvc;
+
+    @Bean
+    public ReferenceGenomeGeneService referenceGenomeGeneService() {
+        return Mockito.mock(ReferenceGenomeGeneService.class);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+
+        Mockito.reset(referenceGenomeGeneService);
+        mockMvc = MockMvcBuilders.webAppContextSetup(wac).build();
+    }
+
+    @Test
+    public void getGene() throws Exception {
+        
+        ReferenceGenomeGene gene = new ReferenceGenomeGene();
+        gene.setEntrezGeneId(ENTREZ_GENE_ID_1);
+        gene.setReferenceGenomeId(REFERENCE_GENOME_ID);
+        gene.setCytoband(CYTOBAND_1);
+        gene.setChromosome(CHROMOSOME_1);
+        gene.setLength(LENGTH_1);
+
+        Mockito.when(referenceGenomeGeneService.getReferenceGenomeGene(Mockito.anyInt(), Mockito.anyString())).thenReturn(gene);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/reference-genome-genes/hg19/1")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.entrezGeneId").value(ENTREZ_GENE_ID_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.referenceGenomeId").value(REFERENCE_GENOME_ID))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cytoband").value(CYTOBAND_1))
```

Comment:
here we are testing that the cytoband differs depending on the reference-genome-gene .. but we are testing that it differs in our mocks, not through the service/persistence layer business logic. So the tests here are fine ... but we still might want some tests at a lower layer showing that the returned cytoband does come from the linked reference_genome_gene (if we don't yet have it)